enterprise/core-edge/src/main/java/org/neo4j/coreedge/raft/state/CoreStateMachines.java

```diff
@@ -154,4 +142,55 @@ public void refresh( TransactionRepresentationCommitProcess localCommit )
 
         coreState.setStateMachine( this, lastAppliedIndex );
     }
+
+    private class StateMachineRequestDispatcher implements RequestDispatcher
```

Comment:
Would be good to try to explain in a comment what is being achieved here. I also wonder if replicatedTxStateMachine really is a regular simple state machine now or if it should be abstracted away under say a BatchedStateMachine abstractions which has the operations queue/sync instead or something like that.